### PR TITLE
Dropdown, TextInput, and Modal Components - Add Missing Props Needed for Faculty Admin Modals

### PR DESCRIPTION
**Pull request by natalynyu**
This PR adds missing properties needed to create the Faculty Admin modal.

`Dropdown`: `disabled` and `required` properties were added for form validation purposes.
`TextInput`: `required` property was added for form validation purposes.
`Modal`: `aria-labelledBy` property was added for accessibility purposes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality
      to change)

## Checklist:

- [x] I have run `eslint` on the code
- [x] I have added JSDoc for all of my code (where applicable)
- [ ] I have added tests to cover my changes.

## Priority:
- [x] Normal 
- [ ] High 

## Related Issues:

Addresses [#214](https://github.com/seas-computing/course-planner/issues/214)

### CODE REVIEW

File: src/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -14,12 +14,14 @@ export interface DropdownProps {
   onChange: ChangeEventHandler;
   /** The name of the dropdown */
   name: string;
-  /** An array of objects with the properties 'label' and 'value' for each dropdown option */
-  options: Array<{label: string; value: string}>;
+  /** An array of objects with the properties 'label', 'value', and 'disabled' for each dropdown option */
+  options: Array<{label: string; value: string; disabled?: boolean}>;
```
Comment (rmainseas) [inline src/Dropdown/Dropdown.tsx:18]: These additional options could do with having their own test cases. For example:

- What happens if I set `options.disabled` to `true`? 
- What happens if I set `options.disabled` to `false`
- What happens if I don't provide a value for `options.disabled`?

File: src/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -14,12 +14,14 @@ export interface DropdownProps {
...
   /** The currently selected dropdown value */
   value?: string;
   /** The dropdown value selected by default */
   defaultValue?: string;
+  /** If true, the dropdown is required to submit the form */
+  required?: boolean;
```
Comment (rmainseas) [inline src/Dropdown/Dropdown.tsx:24]: This could also probably do with a test case

File: src/Modals/Modal.tsx [hunk 3]
```diff
@@ -75,6 +75,10 @@ export interface ModalProps {
    * The content to be displayed within the modal
    */
   children: ReactNode;
+  /**
+   * Identifies the element that provides the modal
+   */
+  ariaLabelledBy: string;
```
Comment (rmainseas) [inline src/Modals/Modal.tsx:81]: I think this is missing a test case

File: examples/Modals/Modal.md [hunk 4]
```diff
@@ -20,6 +20,7 @@ const [modalVisible, setModalVisible] = useState(false);
     Open Modal
   </Button>
   <Modal
+    ariaLabelledBy="testButton"
```
Comment (rmainseas) [inline examples/Modals/Modal.md:23]: There is no `testButton` in this example

File: examples/Modals/Modal.md [hunk 5]
```diff
@@ -59,6 +60,7 @@ const [formValue, setFormValue] = useState('');
...
```
Comment (rmainseas) [inline examples/Modals/Modal.md:63]: See above RE: `testButton`

File: src/Dropdown/Dropdown.tsx (hunk 1 above)
Comment (rmainseas) [inline src/Dropdown/Dropdown.tsx:18]: I would extract this type out on it's own, or better yet - type this as [`HTMLOption[]`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOptionElement)

Comment (natalynyu) [reply src/Dropdown/Dropdown.tsx:18]: Good point! I added tests through 03a9141.

File: examples/Modals/Modal.md (hunk 4 above)
Comment (natalynyu) [reply examples/Modals/Modal.md:26]: Ah, my mistake. I updated this through a0c278a.

File: examples/Modals/Modal.md (hunk 5 above)
Comment (natalynyu) [reply examples/Modals/Modal.md:69]: I updated this as well through a0c278a.

File: src/Modals/Modal.tsx (hunk 3 above)
Comment (natalynyu) [reply src/Modals/Modal.tsx:81]: I added a test to check the value of the aria-labelledBy property through commit 0a12491.

File: src/Dropdown/Dropdown.tsx (hunk 2 above)
Comment (natalynyu) [reply src/Dropdown/Dropdown.tsx:24]: Good point - I added a test case for this through 1fa513b.

File: src/Dropdown/Dropdown.tsx (hunk 1 above)
Comment (natalynyu) [reply src/Dropdown/Dropdown.tsx]: Ok good idea - I extracted this type out through 2f0f88b.

Comment (natalynyu) [reply src/Dropdown/Dropdown.tsx]: When I tried to type it as `HTMLOption`, it made all of the properties under HTMLOption mandatory, so I ended up extracting the type out instead

Comment (rmainseas) [reply src/Dropdown/Dropdown.tsx]: Ah, that's frustrating. Fair enough :)

File: src/Dropdown/__tests__/Dropdown.test.tsx [hunk 6]
```diff
@@ -43,19 +53,51 @@ describe('Dropdown', function () {
   });
   it('contains the expected elements', function () {
     const dropdownOptionsCount = getAllByRole('option').length;
-    strictEqual(dropdownOptionsCount, 3);
+    strictEqual(dropdownOptionsCount, 4);
```
Comment (rmainseas) [inline src/Dropdown/__tests__/Dropdown.test.tsx:56]: You might want to compare this with `options.length`, rather than a magic number :)

Comment (natalynyu) [reply src/Dropdown/__tests__/Dropdown.test.tsx]: Ah! Good point. I updated this through commit c29d078.

**Pull request merged**